### PR TITLE
Page content backend

### PR DESCRIPTION
**Pull request by MihneaDinuGL**
The backend for the diagram component.

### CODE REVIEW

File: src/main/java/com/globallogic/amcr/service/pagecontent/DiagramServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        this.diagramDao = diagramDao;
+    }
+
+    @Transactional
+    public void save(Diagram diagram) {
+        try {
+            UUID diagramId = UUID.randomUUID();
+            diagramDao.save(diagram, diagramId);
+        } catch (Exception e) {
+            throw new RuntimeException("Error saving in service implementation", e);
```
Comment (thomas-ayling) [inline src/main/java/com/globallogic/amcr/service/pagecontent/DiagramServiceImpl.java:30]: More detail on which method the error was thrown could be useful for debugging

File: src/main/java/com/globallogic/amcr/service/pagecontent/DiagramServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    @Transactional(readOnly = true)
+    public Diagram getByNode(int nodeId) { return diagramDao.getByNode(nodeId);}
+    @Transactional(readOnly = true)
+    public List<Diagram> getAll() {return diagramDao.getAll();}
+
+    @Transactional
+    public void update(Diagram diagram, UUID id) {
+        try {
+            diagramDao.update(diagram, id);
```
Comment (thomas-ayling) [inline src/main/java/com/globallogic/amcr/service/pagecontent/DiagramServiceImpl.java:45]: Make sure that the received object is not null

**Review by thomas-ayling — CHANGES_REQUESTED** [review]

File: src/main/java/com/globallogic/amcr/persistence/model/pagecontent/Diagram.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    public Diagram() {
+    }
+
+    public UUID getId() {
+        return id;
+    }
+
+    public void setId(UUID id) {
+        Assert.notNull(id, "ID is null");
```
[the remark this answers is not in the document]
Comment (thomas-ayling) [reply src/main/java/com/globallogic/amcr/persistence/model/pagecontent/Diagram.java:42]: That's outdated now, you can use `Assert.assertNull()` instead

File: src/main/java/com/globallogic/amcr/mapper/pagecontent/DiagramMapper.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
+package com.globallogic.amcr.mapper.pagecontent;
```
[the remark this answers is not in the document]
Comment (MihneaDinuGL) [reply src/main/java/com/globallogic/amcr/mapper/pagecontent/DiagramMapper.java:1]: Is this something everyone should take into account?

File: src/main/java/com/globallogic/amcr/persistence/dao/pagecontent/DiagramDao.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    /**
+     * @param diagram the diagram object received from client, no ID set
+     * @param diagramId the ID for the diagram object
+     */
+    @Override
+    public void save(Diagram diagram, UUID diagramId) {
+        try {
+            diagram.setId(diagramId);
+            diagramMapper.save(diagram);
+        } catch (Exception e){
```
[the remark this answers is not in the document]
Comment (MihneaDinuGL) [reply src/main/java/com/globallogic/amcr/persistence/dao/pagecontent/DiagramDao.java:26]: What kind of exception would be expected to be thrown here? I will change the message so it's more specific as well

[the remark this answers is not in the document]
Comment (MihneaDinuGL) [reply src/main/java/com/globallogic/amcr/persistence/dao/pagecontent/DiagramDao.java:26]: I have tried to be more specific about the error, but I'd be happy to remove it if you consider that better

**Review by Kya-Edmeade — APPROVED** [review]

**Review by thomas-ayling — APPROVED** [review]

**Pull request merged**